src/widgets/category/category-widget-model.js

```diff
@@ -55,16 +53,31 @@ module.exports = WidgetModel.extend({
     this.lockedCategories.reset([]);
   },
 
-  applyColors: function () {
-    this.set('isColorsApplied', true);
+  autoStyle: function () {
+    var style = this.autoStyler.getStyle();
+    if (!this.dataviewModel._dataProvider) {
```

Comment:
IMO we shouldn't access `_dataProvider` here... It's supposed be a private thing. My suggestion would be to remove the "if" and just set the new styles on the layer, letting CartoDB.js take care of "everything" else internally. Something like:

```
var style = this.autoStyler.getStyle();
this.dataviewModel.layer.set('cartocss', style);
this.dataviewModel.set('autoStyle', true); // Not sure how this is being used
```

There's already [this binding](https://github.com/CartoDB/cartodb.js/blob/v4/src/geo/leaflet/leaflet-cartodb-vector-layer-group-view.js#L25-L28), wouldn't that update the styles on the "vector sublayer"?